src/app/projects/edit-project/edit-project.component.spec.ts

```diff
@@ -0,0 +1,124 @@
+import { async, ComponentFixture, TestBed } from '@angular/core/testing';
+
+import { EditProjectComponent } from './edit-project.component';
+import { MAT_BOTTOM_SHEET_DATA, MatBottomSheetRef } from '@angular/material';
+import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
+import { LayoutModule } from '@angular/cdk/layout';
+import { FlexLayoutModule } from '@angular/flex-layout';
+import { ReactiveFormsModule } from '@angular/forms';
+import { AppMaterialModule } from '../../app-material.module';
+import { GlobalErrorHandlerService } from '../../error/global-error-handler.service';
+import { Observable, of } from 'rxjs';
+import { ProjectsService } from '../projects.service';
+import { Project } from '../project';
+
+const bottomSheetStub: Partial<MatBottomSheetRef> = {
+  dismiss(result?: any): void { }
+};
+
+const projectsServiceStub: Partial<ProjectsService> = {
+  updateProject(project: Project):
+    Observable<Project> { return null; }
+};
+
+const projectTestData: Project = {
+  id: 'e6b808eb-b6bd-447d-8dce-3e0d66b17759',
+  name: 'Project name',
+  customer: 'Customer',
+  industrySector: 'Industry sector',
+  description: 'Description',
+} as Project;
+
+describe('EditProjectComponent', () => {
+  let component: EditProjectComponent;
+  let fixture: ComponentFixture<EditProjectComponent>;
+
+  beforeEach(async(() => {
+    spyOn(projectsServiceStub, 'updateProject')
+      .and.returnValue(of<Project>(
+      {
+        id: 'e6b808eb-b6bd-447d-8dce-3e0d66b17759',
+        name: 'Changed name',
+        customer: 'Changed customer',
+        industrySector: 'Changed industrySector',
+        description: 'Changed description',
+        creationDate: new Date(),
+        lastModifiedDate: new Date()
+      }));
+    TestBed.configureTestingModule({
+      imports: [
+        BrowserAnimationsModule,
+        LayoutModule,
+        FlexLayoutModule,
+        ReactiveFormsModule,
+        AppMaterialModule
+      ],
+      declarations: [ EditProjectComponent ],
+      providers: [
+        GlobalErrorHandlerService,
+        { provide: ProjectsService, useValue: projectsServiceStub },
+        { provide: MatBottomSheetRef, useValue: bottomSheetStub },
```

Comment:
I would suggest to use here `jasmine.createSpyObj('matBottomSheetRef', ['dismiss'] )` instead of bottomSheetStub.
The reasons for that are usage of a special jasmine-testing API and reducing of number of lines of source code.
Sveta, is it OK for you?

Reply:
done